apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+---
+title: Package Configuration
+layout: learn
+authors: JakobJingleheimer
+---
+
+# Package configuration
+
+Configuration is always a chore, but an unfortunately necessary evil. And configuring a package for CommonJS (CJS) and ES Modules (ESM) can be a waking nightmare—not least because it has changed a dozen times in half as many years.
+
+A frequent question is “how do I make this work!?” (often with angry tears); but yet more frequently we come across packages that are just misconfigured.
+
+All the provided `package.json` configurations (not specifically marked “does not work”) work in Node.js 12.22.x (v12 latest, the oldest supported line) and 17.2.0 (current latest at the time)[^1], and for grins, with webpack 5.53.0 and 5.63.0 respectively. These are available: [nodejs/package-examples](https://github.com/nodejs/package-examples/blob/main/config).
+
+For curious cats, [Preamble: How did we get here](#preamble-how-did-we-get-here) and [Down the rabbit-hole](#down-the-rabbithole) provide background and deeper explanations. If you're just looking for a solution, jump to [Pick your fix](#pick-your-fix) for the TLDR.
+
+## General notes
+
+[Syntax detection](https://nodejs.org/api/packages.html#syntax-detection) is _**not**_ a replacement for proper package configuration; syntax detection is not fool-proof and it has [significant performance cost](https://github.com/nodejs/node/pull/55238).
```

Comment:
I don't recall asking about syntax detection per-se, but generally I think it should be an implementation detail that package authors rarely know about as for anyone who has a `package.json`, the most no-brainer thing to do is to just use explicit type fields. The article assumes that readers know what even is syntax detection, but I have seen several experienced maintainers not even knowing it's a thing, let alone beginners...

Reply:
> I don't recall asking about syntax detection per-se

True, you had mentioned it in regard to other things. This seemed in the same general FYI vein.

If they don't know what syntax detection is though, is that a problem? They can safely ignore it (I could add a note like "if you don't know what this is, don't worry about it"); whereas someone who is aware of it could benefit from the forewarning.

Reply:
We talked about information flow during the collab summit for beginners, and I think the issue with throwing details like this in the beginning of a tutorial is that it makes the whole thing a lot more intimidating for beginners. The ideal flow I think for such a guide is:

1. If you author in CJS, ship CJS and this is how you do it.
2. If you author in ESM, ship ESM and this is how you do it if you don't care about EOL Node.js versions

At this point, >80% of the reader should get what they are looking for in the tutorial and can already leave. And the remaining >20% can continue reading about the legacy practices for supporting EOL Node.js versions and hacky ways to transpile CJS into ESM and all their pitfalls and gotchas. People who need to know how to do other methods have to know how 1 and 2 is done to get some basics going anyway, but for people who only need to know about 1 and 2 the others are just some legacy noise or hacks they may never need to know about.

Reply:
Hmm, that sounds like a _major_ restructuring of this huge article. Do you have a specific proposal?